### PR TITLE
chore(deps): update go from 1.20.5 to 1.20.6

### PR DESCRIPTION
**Pull request by slonka**
same as https://github.com/kumahq/kuma/pull/7330 but with testcontainers update to main

### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by jakubdyszkiewicz — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -50,7 +50,7 @@ require (
 	github.com/spf13/cobra v1.7.0
 	github.com/spf13/viper v1.16.0
 	github.com/spiffe/go-spiffe/v2 v2.1.6
-	github.com/testcontainers/testcontainers-go v0.21.0
+	github.com/testcontainers/testcontainers-go v0.21.1-0.20230802082724-c0797a06f2c2 // UPDATE TO RELEASE TAG ASAP
```
Comment (slonka) [inline go.mod:53]: FYI

**Review by michaelbeaumont — APPROVED** [review]

**Review by bartsmykla — APPROVED** [review]

File: mk/dependencies/deps.lock [hunk 2]
```diff
@@ -1 +1 @@
-ddcb18fef158bd55f6534f1a4ab75af4f9db3b6f
+3bee12a0f3d1e3c04bb344135cf5a624a3feedc9
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply mk/dependencies/deps.lock:1]: `mk/dev.mk` goes into the hash

[the remark this answers is not in the document]
Comment (slonka) [reply mk/dependencies/deps.lock:1]: 👍

**Pull request merged**